File: Algorithms/p205_Isomorphic_Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n = s.size();
        vector<int> mapping(256,-1);
        vector<bool> mapped(256,0);
        for(int i=0;i<n;++i){
            auto c1 = s[i];
            auto c2 = t[i];
            if(mapping[c1]==-1){
                mapping[c1] = c2;
                if(!mapped[c2]) mapped[c2] = 1;
                else return 0;
            }else if(mapping[c1]!=c2){
                return 0;
            }
        }
        return 1;
    }
};

File: Algorithms/p205_Isomorphic_Strings/p205_Isomorphic_Strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        mapping = {chr(i):None for i in range(256)}
        mapped = {chr(i):False for i in range(256)}
        for c1,c2 in zip(s,t):
            if mapping[c1]==None:
                mapping[c1] = c2
                if not mapped[c2]:
                    mapped[c2] = True
                else:
                    return False
            elif mapping[c1]!=c2:
                return False
        return True
